src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+     * (Lsb0) bit ordering.
+     * </p>
+     * <p>
+     * (1, 0, 0, 0) is converted as follow: '1'
+     * </p>
+     *
+     * @param src the binary to convert
+     * @param srcPos the position of the lsb to start the conversion
+     * @return a hexadecimal digit representing the selected bits
+     * @throws IllegalArgumentException if {@code src} is empty
+     * @throws NullPointerException if {@code src} is {@code null}
+     */
+    public static char binaryToHexDigit(final boolean[] src, final int srcPos) {
+        if (src.length == 0) {
+            throw new IllegalArgumentException("Cannot convert an empty array.");
+        }
+        if (src.length > srcPos + 3 && src[srcPos + 3]) {
+            if (src[srcPos + 2]) {
+                if (src[srcPos + 1]) {
+                    return src[srcPos] ? 'f' : 'e';
+                }
+                return src[srcPos] ? 'd' : 'c';
+            }
+            if (src[srcPos + 1]) {
+                return src[srcPos] ? 'b' : 'a';
+            }
+            return src[srcPos] ? '9' : '8';
+        }
+        if (src.length > srcPos + 2 && src[srcPos + 2]) {
+            if (src[srcPos + 1]) {
+                return src[srcPos] ? '7' : '6';
+            }
+            return src[srcPos] ? '5' : '4';
+        }
+        if (src.length > srcPos + 1 && src[srcPos + 1]) {
+            return src[srcPos] ? '3' : '2';
+        }
+        return src[srcPos] ? '1' : '0';
+    }
+
+    /**
+     * <p>
+     * Converts binary (represented as boolean array) to a hexadecimal digit using the Msb0 bit
+     * ordering.
+     * </p>
+     * <p>
+     * (1, 0, 0, 0) is converted as follow: '8'
+     * </p>
+     *
+     * @param src the binary to convert
+     * @return a hexadecimal digit representing the selected bits
+     * @throws IllegalArgumentException if {@code src} is empty, {@code src.length < 4} or
+     *             {@code src.length > 8}
+     * @throws NullPointerException if {@code src} is {@code null}
+     */
+    public static char binaryToHexDigitMsb0_4bits(final boolean[] src) {
+        return binaryToHexDigitMsb0_4bits(src, 0);
+    }
+
+    /**
+     * <p>
+     * Converts binary (represented as boolean array) to a hexadecimal digit using the Msb0 bit
+     * ordering.
+     * </p>
+     * <p>
+     * (1, 0, 0, 0) is converted as follow: '8' (1, 0, 0, 1, 1, 0, 1, 0) with srcPos = 3 is converted
+     * to 'D'
+     * </p>
+     *
+     * @param src the binary to convert
+     * @param srcPos the position of the lsb to start the conversion
+     * @return a hexadecimal digit representing the selected bits
+     * @throws IllegalArgumentException if {@code src} is empty, {@code src.length > 8} or
+     *             {@code src.length - srcPos < 4}
+     * @throws NullPointerException if {@code src} is {@code null}
+     */
+    public static char binaryToHexDigitMsb0_4bits(final boolean[] src, final int srcPos) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 380-391, lines 392-397 into separate methods.